### PR TITLE
add Scientific notation of float

### PR DESCRIPTION
**Pull request by czpmango**

### CODE REVIEW

File: src/parser/scanner.lex [hunk 1]
```diff
@@ -334,7 +335,9 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
                             }
 
 {DEC}*\.{DEC}+              |
-{DEC}+\.{DEC}*              {
+{DEC}+\.{DEC}*              |
+{DEC}+\.{DEC}*{EXP}         |
+{DEC}+{EXP}                 {
                                 return parseDouble();
```
Comment (Shylock-Hg) [inline src/parser/scanner.lex:341]: How about integer type?

File: src/parser/scanner.lex [hunk 2]
```diff
@@ -334,7 +335,9 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
                             }
 
 {DEC}*\.{DEC}+              |
-{DEC}+\.{DEC}*              {
+{DEC}+\.{DEC}*              |
+{DEC}+\.{DEC}*{EXP}         |
+{DEC}+{EXP}                 {
```
Comment (jievince) [inline src/parser/scanner.lex:340]:
```
.3e4

**Review by nevermore3 — DISMISSED** [review]

File: src/parser/scanner.lex [hunk 3]
```diff
@@ -33,6 +33,7 @@ NOT_ENDS_WITH               (NOT{blanks}ENDS{blanks}WITH)
 
 LABEL                       ([a-zA-Z][_a-zA-Z0-9]*)
 DEC                         ([0-9])
+EXP                         ([eE][-+]?[0-9]*)
```
Comment (jievince) [inline src/parser/scanner.lex:36]: should be `+`?

File: src/parser/scanner.lex (hunk 1 above)
Comment (jievince) [reply src/parser/scanner.lex:341]: > How about integer type?

1e5 is parsed as float, not an integer.
image

File: src/parser/scanner.lex (hunk 2 above)
Comment (czpmango) [reply src/parser/scanner.lex:341]: Fixed.

File: src/parser/scanner.lex (hunk 3 above)
Comment (czpmango) [reply src/parser/scanner.lex]: ACK.

File: src/parser/scanner.lex (hunk 1 above)
Comment (czpmango) [reply src/parser/scanner.lex:342]: `2e3`?

File: src/parser/scanner.lex [hunk 4]
```diff
@@ -334,7 +335,10 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
                             }
 
 {DEC}*\.{DEC}+              |
-{DEC}+\.{DEC}*              {
+{DEC}+\.{DEC}*              |
+\.{DEC}*{EXP}               |
+{DEC}+\.{DEC}*{EXP}         |
```
Comment (jievince) [inline src/parser/scanner.lex:340]: can comebine into one rule:
```
{DEC}*\.{DEC}* 
```

**Review by jievince — APPROVED** [review]

File: src/parser/scanner.lex (hunk 4 above)
Comment (czpmango) [reply src/parser/scanner.lex:342]: ok

**Review by jievince — DISMISSED** [review]

File: tests/tck/features/yield/yield.feature [hunk 5]
```diff
@@ -6,11 +6,11 @@ Feature: Yield Sentence
   Scenario: Base
     When executing query:
       """
-      YIELD 1
+      YIELD [1, 1.1, 1e2, 1.1e2, .3e4, 1.e4, true] AS basic_value
```
Comment (yixinglu) [inline tests/tck/features/yield/yield.feature:9]: `1234E-10`

**Review by yixinglu — APPROVED** [review]

**Review by laura-ding — APPROVED** [review]

**Pull request merged**